### PR TITLE
Features for hub

### PR DESCRIPTION
**Pull request by AnkitCLI**
This PR contains feature files for Hub sanity e2e tests.

### CODE REVIEW

File: cdap-e2e-tests/src/e2e-test/java/io/cdap/cdap/hub/stepsdesign/TestSetupHooks.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      throws IOException, URISyntaxException {
+    String bucketName = StorageClient.createBucket("e2e-test-" + UUID.randomUUID()).getName();
+    StorageClient.uploadObject(bucketName, filePath, filePath);
+    return bucketName;
+  }
+
+  @After(order = 1, value = "@GCS_SOURCE_TEST")
+  public static void deleteSourceBucketWithFile() {
+    deleteGCSBucket(gcsSourceBucketName);
+    gcsSourceBucketName = StringUtils.EMPTY;
```
Comment (Aryan-Verma) [inline cdap-e2e-tests/src/e2e-test/java/io/cdap/cdap/hub/stepsdesign/TestSetupHooks.java]: Why is this needed?

File: cdap-e2e-tests/src/e2e-test/java/io/cdap/cdap/hub/stepsdesign/TestSetupHooks.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    PluginPropertyUtils.addPluginProp("gcsTargetPath", "gs://" + gcsTargetBucketName);
+    BeforeActions.scenario.write("GCS target bucket name - " + gcsTargetBucketName);
+  }
+
+  @After(order = 1, value = "@GCS_SINK_TEST")
+  public static void deleteTargetBucketWithFile() {
+    deleteGCSBucket(gcsTargetBucketName);
+    PluginPropertyUtils.removePluginProp("gcsTargetBucketName");
+    PluginPropertyUtils.removePluginProp("gcsTargetPath");
+    gcsTargetBucketName = StringUtils.EMPTY;
```
Comment (Aryan-Verma) [inline cdap-e2e-tests/src/e2e-test/java/io/cdap/cdap/hub/stepsdesign/TestSetupHooks.java]: Why is this needed?

**Review by Aryan-Verma — COMMENTED** [review]
Where is the stepsdesign for these steps? Is there a separate PR for the same?

File: cdap-e2e-tests/src/e2e-test/java/io/cdap/cdap/hub/stepsdesign/TestSetupHooks.java (hunk 1 above)
Comment (AnkitCLI) [reply cdap-e2e-tests/src/e2e-test/java/io/cdap/cdap/hub/stepsdesign/TestSetupHooks.java]: We need hooks for creating and deleting gcs bucket because in one of the feature we are creating  a pipeline from hub and in order to customize and deploy that pipeline we need to use these hooks because in that pipeline we have file and gcs plugin as source and sink so we have to use gcs path in that.

File: cdap-e2e-tests/src/e2e-test/java/io/cdap/cdap/hub/stepsdesign/TestSetupHooks.java (hunk 2 above)
Comment (AnkitCLI) [reply cdap-e2e-tests/src/e2e-test/java/io/cdap/cdap/hub/stepsdesign/TestSetupHooks.java]: Same as above.

**Comment by AnkitCLI** [comment]
> Where is the stepsdesign for these steps? Is there a separate PR for the same?

Yes we have raised a separate PR for Locators,Stepdesign and action classes at e2e framework level. https://github.com/cdapio/cdap-e2e-tests/pull/230

File: cdap-e2e-tests/src/e2e-test/java/io/cdap/cdap/hub/stepsdesign/TestSetupHooks.java (hunk 1 above)
Comment (Aryan-Verma) [reply cdap-e2e-tests/src/e2e-test/java/io/cdap/cdap/hub/stepsdesign/TestSetupHooks.java]: I meant to say that we don't need this line:
`gcsSourceBucketName = StringUtils.EMPTY;`
We anyways assign it a new name in the Before hook every time.

Comment (AnkitCLI) [reply cdap-e2e-tests/src/e2e-test/java/io/cdap/cdap/hub/stepsdesign/TestSetupHooks.java]: done

File: cdap-e2e-tests/src/e2e-test/java/io/cdap/cdap/hub/stepsdesign/TestSetupHooks.java (hunk 2 above)
Comment (AnkitCLI) [reply cdap-e2e-tests/src/e2e-test/java/io/cdap/cdap/hub/stepsdesign/TestSetupHooks.java]: done

**Review by Aryan-Verma — APPROVED** [review]

**Review by DJSagarAhire — APPROVED** [review]
LGTM
